seisbench/models/base.py

```diff
@@ -595,34 +683,67 @@ def annotate(
                                                This is usually acceptable for rotationally invariant models,
                                                e.g., most picking models.
         :type flexible_horizontal_components: bool
+        :param parallelism: If None, uses the `asyncio` implementation. Otherwise, defines the redundancy for each
+                            subjob, i.e., parallelism=2 would start each subjob twice. See the warning above for a
+                            discussion on parallelism for annotate.
+        :type parallelism: None, int
         :param kwargs:
         :return: Obspy stream of annotations
         """
-        nest_asyncio.apply()
-        call = self._annotate_async(
-            stream, strict, flexible_horizontal_components, **kwargs
+        # nest_asyncio.apply()
+        self._check_parallelism_annotate(stream, parallelism)
+
+        if parallelism is None:
+            nest_asyncio.apply()
+            call = self._annotate_async(
+                stream, strict, flexible_horizontal_components, **kwargs
+            )
+            return asyncio.run(call)
+        else:
+            return self._annotate_processes(
+                stream, strict, flexible_horizontal_components, **kwargs
+            )
+
+    @staticmethod
+    def _check_parallelism_annotate(stream, parallelism, thresholds=(5e5, 5e7)):
+        """
+        Checks whether the chosen parallelism looks reasonable and prints a warning otherwise.
+
+        :param stream: Data stream
+        :type stream: obspy.Stream
+        :param parallelism: parallelism indicator
+        :type parallelism: None, int
+        :return: None
+        """
```

Comment:
I really like this helper method! As a quick general question from my side, were these thresholds chosen arbitrarily or is this where the runtime performance becomes quicker for multiprocessing over asyncio from some quick exploratory tests?

Reply:
I did a few quick tests to choose these values. But as you see, I added quite a big margin in between the lower and upper bound. I think the actual break-even points will depend on a lot of factors, e.g., the operating system or the model. For example, DeepDenoiser has pretty expensive pre-/post-processing (FFTs). Consequently, for DeepDenoiser using parallelism pays of by far earlier than, for example, for PhaseNet.

Reply:
Yeah true - good point. I think as a general guide this is a great addition as a reminder/pointer that we have this functionality now available.